Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -3702,18 +3702,13 @@ internal unsafe bool UpdateState(InputDevice device, InputEvent* eventPtr, Input
             ////REVIEW: Should we do this only for events but not for InputState.Change()?
             // If noise filtering on .current is turned on and the device may have noise,
             // determine if the event carries signal or not.
-            var makeDeviceCurrent = true;
-            if (device.noisy && m_Settings.filterNoiseOnCurrent)
-            {
-                // Compare the current state of the device to the newly received state but overlay
-                // the comparison by the noise mask.
-
-                var noiseMask = (byte*)InputStateBuffers.s_NoiseMaskBuffer + deviceStateOffset;
-
-                makeDeviceCurrent =
-                    !MemoryHelpers.MemCmpBitRegion(deviceStatePtr, statePtr,
-                        0, stateSize * 8, mask: noiseMask);
-            }
+            var noiseMask = device.noisy && m_Settings.filterNoiseOnCurrent
+                ? (byte*)InputStateBuffers.s_NoiseMaskBuffer + deviceStateOffset
+                : null;
+            // Compare the current state of the device to the newly received state but overlay
+            // the comparison by the noise mask.
+            var makeDeviceCurrent = !MemoryHelpers.MemCmpBitRegion(deviceStatePtr, statePtr,
+                0, stateSize * 8, mask: noiseMask);
```

Comment:
Hmm, I wonder whether we should `if` the check here behind `m_Settings.filterNoiseOnCurrent`. I.e. like

```
var makeDeviceCurrent = true;
if (m_Settings.filterNoiseOnCurrent)
{
            var noiseMask = device.noisy && m_Settings.filterNoiseOnCurrent
                ? (byte*)InputStateBuffers.s_NoiseMaskBuffer + deviceStateOffset
                : null;
            // Compare the current state of the device to the newly received state but overlay
            // the comparison by the noise mask.
            makeDeviceCurrent = !MemoryHelpers.MemCmpBitRegion(deviceStatePtr, statePtr,
                0, stateSize * 8, mask: noiseMask);
}
```

The intention behind `filterNoiseOnCurrent` is to disable any kind of filtering on `current`.

Reply:
`filterNoiseOnCurrent` is false by default though, maybe we can make it true by default and then I can wrap whole machinery under this setting? e.g. I mean I want memcmp checks to be on by default